File: JavaDemo/AndroidDemo/app/src/main/java/com/example/androiddemo/NetworkThread.java
package com.example.androiddemo;

import java.io.IOException;
import java.net.HttpURLConnection;
import java.net.URL;
import java.net.URLEncoder;
import java.nio.charset.StandardCharsets;
import java.util.concurrent.Callable;

public class NetworkThread implements Callable<String> {
    private final String name;
    private final String password;
    private final String url;

    public NetworkThread(String name, String password, String url) {
        this.name = name;
        this.password = password;
        this.url = url;
    }

    @Override
    public String call(){
        try {
            HttpURLConnection connection = (HttpURLConnection) new URL(url).openConnection();
            String data = "name="+ URLEncoder.encode(name, StandardCharsets.UTF_8.toString())+"&password="+URLEncoder.encode(password,StandardCharsets.UTF_8.toString());
            connection.setRequestMethod("POST");
            connection.setDoInput(true);
            connection.setDoOutput(true);
            connection.getOutputStream().write(data.getBytes(StandardCharsets.UTF_8));
            byte [] bytes = new byte[1024];
            int len = connection.getInputStream().read(bytes);
            return new String(bytes,0,len,StandardCharsets.UTF_8);
        } catch (IOException e) {
            e.printStackTrace();
            return "";
        }
    }
}

File: KotlinDemo/AndroidDemo/app/src/main/java/com/example/androiddemo/NetworkThread.kt
package com.example.androiddemo

import java.lang.Exception
import java.net.HttpURLConnection
import java.net.URL
import java.net.URLEncoder
import java.util.concurrent.Callable

class NetworkThread(var name:String,var password:String,var url:String):Callable<String> {
    override fun call(): String {
        return try{
            val connection = URL(url).openConnection() as HttpURLConnection
            val data = "name="+URLEncoder.encode(name,Charsets.UTF_8.toString())+"&password="+URLEncoder.encode(password,Charsets.UTF_8.toString())
            connection.requestMethod = "POST"
            connection.doInput = true
            connection.doOutput = true
            connection.outputStream.write(data.toByteArray(Charsets.UTF_8))
            val bytes = ByteArray(1024)
            val len = connection.inputStream.read(bytes)
            String(bytes,0,len,Charsets.UTF_8)
        }catch (e:Exception){
            e.printStackTrace()
            ""
        }
    }
}